File: backlands-client/src/network/opcodes.js
// Client → Server
export const C_LOGIN       = 0x01;
export const C_ENTER_GAME  = 0x0F;
export const C_LOGOUT      = 0x14;
export const C_PING        = 0x1D;
export const C_WALK_NORTH  = 0x65;
export const C_WALK_EAST   = 0x66;
export const C_WALK_SOUTH  = 0x67;
export const C_WALK_WEST   = 0x68;
export const C_TURN_NORTH  = 0x6F;
export const C_TURN_EAST   = 0x70;
export const C_TURN_SOUTH  = 0x71;
export const C_TURN_WEST   = 0x72;
export const C_ATTACK             = 0xA1;
export const C_CHANGE_FIGHT_MODES = 0xA0;
export const C_TALK               = 0x96;

// Server → Client
export const S_LOGIN_ERROR    = 0x0A;
export const S_ENTER_GAME     = 0x17;
export const S_PING_BACK      = 0x1D;
export const S_PING           = 0x1E;
// 0x64: S_CHAR_LIST in login phase / S_FULL_MAP in game phase (TFS dual-port convention)
export const S_CHAR_LIST      = 0x64;
export const S_FULL_MAP       = 0x64;
export const S_MAP_TOP_ROW    = 0x65;
export const S_MAP_RIGHT_COL  = 0x66;
export const S_MAP_BOTTOM_ROW = 0x67;
export const S_MAP_LEFT_COL   = 0x68;
export const S_CREATE_ON_MAP  = 0x6A;
export const S_DELETE_ON_MAP  = 0x6C;
export const S_MOVE_CREATURE  = 0x6D;
export const S_PLAYER_DATA    = 0xA0;
export const S_TEXT_MESSAGE   = 0xB4;
export const S_TALK             = 0xAA;
export const S_CREATURE_HEALTH  = 0x8C;
export const S_GRAPHICAL_EFFECT = 0x83;
export const S_TEXT_EFFECT      = 0x84;
export const S_MISSILE_EFFECT   = 0x85;
export const S_DEATH            = 0x28;

File: backlands-server/src/network/opcodes.rs
// Client → Server opcodes (TFS-aligned)
pub const C_LOGIN:      u8 = 0x01;
pub const C_ENTER_GAME: u8 = 0x0F;
pub const C_LOGOUT:     u8 = 0x14;
pub const C_PING:       u8 = 0x1D;
pub const C_WALK_NORTH: u8 = 0x65;
pub const C_WALK_EAST:  u8 = 0x66;
pub const C_WALK_SOUTH: u8 = 0x67;
pub const C_WALK_WEST:  u8 = 0x68;
pub const C_TURN_NORTH: u8 = 0x6F;
pub const C_TURN_EAST:  u8 = 0x70;
pub const C_TURN_SOUTH: u8 = 0x71;
pub const C_TURN_WEST:  u8 = 0x72;

// Server → Client opcodes (TFS-aligned)
pub const S_LOGIN_ERROR:    u8 = 0x0A;
pub const S_ENTER_GAME:     u8 = 0x17;
pub const S_PING_BACK:      u8 = 0x1D;
pub const S_PING:           u8 = 0x1E;
// 0x64: S_CHAR_LIST in login phase / S_FULL_MAP in game phase (same opcode, different ports in TFS)
pub const S_CHAR_LIST:      u8 = 0x64;
pub const S_FULL_MAP:       u8 = 0x64;
pub const S_MAP_TOP_ROW:    u8 = 0x65; // player walked north  — send new top row
pub const S_MAP_RIGHT_COL:  u8 = 0x66; // player walked east   — send new right col
pub const S_MAP_BOTTOM_ROW: u8 = 0x67; // player walked south  — send new bottom row
pub const S_MAP_LEFT_COL:   u8 = 0x68; // player walked west   — send new left col
pub const S_CREATE_ON_MAP:  u8 = 0x6A; // TFS: GameServerCreateOnMap — spawn creature
pub const S_DELETE_ON_MAP:  u8 = 0x6C; // TFS: GameServerDeleteOnMap — despawn creature
pub const S_MOVE_CREATURE:  u8 = 0x6D; // TFS: GameServerMoveCreature — creature moved
pub const S_PLAYER_DATA:    u8 = 0xA0;
pub const S_TEXT_MESSAGE:   u8 = 0xB4;
